builder/comp-builder.nix

```diff
@@ -39,6 +39,7 @@ let self =
                # ghc's internal linker seems to be broken on x86.
                && !(stdenv.hostPlatform.isMusl && !stdenv.hostPlatform.isx86)
 , enableDeadCodeElimination ? component.enableDeadCodeElimination
+, writeHieFiles ? component.writeHieFiles
```

Comment:
`enableHie`? `doHie`? Our option names are pretty inconsistent, but it seems like it should either be like `enableShared` or `doHaddock`...

Reply:
Yea, this is not a great name - it doesn't even map to the GHC option... Happy to defer to a maintainer to make the call here.

Reply:
I nominate @angerman :p

Reply:
enableHie seems good to me.